### PR TITLE
feat: replace provisioning-admins group-based permissions with role-based

### PR DESCRIPTION
**Pull request by hamzawaleed01**
## Post-review

* [x] Squash commits into discrete sets of changes
* [x] Ensure that once the changes have been [deployed to stage](https://gocd.tools.edx.org/go/pipeline/activity/stage-enterprise_catalog), prod is manually deployed

### CODE REVIEW

File: enterprise_catalog/apps/catalog/rules.py [hunk 1]
```diff
@@ -102,6 +104,25 @@ def has_explicit_access_to_catalog_learner(user, context):
...
+    """
+    request = crum.get_current_request()
+    decoded_jwt = get_decoded_jwt(request) or get_decoded_jwt_from_auth(request)
+    return request_user_has_implicit_access_via_jwt(decoded_jwt, ENTERPRISE_CATALOG_PROVISIONING_ADMIN_ROLE, context)
+
+
+rules.add_perm(
+    PERMISSION_HAS_CATALOG_PROVISIONING_ADMIN_ACCESS,
+    has_implicit_access_to_enterprise_catalog_provisioining
+)
```
Comment (sameenfatima78) [inline enterprise_catalog/apps/catalog/rules.py]: Question: Why didn't we add a rule for explicit access?

File: enterprise_catalog/apps/catalog/constants.py [hunk 2]
```diff
@@ -63,16 +63,19 @@
...
 SYSTEM_ENTERPRISE_LEARNER_ROLE = 'enterprise_learner'
 SYSTEM_ENTERPRISE_ADMIN_ROLE = 'enterprise_admin'
 SYSTEM_ENTERPRISE_OPERATOR_ROLE = 'enterprise_openedx_operator'
+SYSTEM_ENTERPRISE_PROVISIONING_ADMIN_ROLE = 'enterprise_provisioning_admin'
 
 ACCESS_TO_ALL_ENTERPRISES_TOKEN = '*'
 
 PERMISSION_HAS_LEARNER_ACCESS = 'catalog.has_learner_access'
 PERMISSION_HAS_ADMIN_ACCESS = 'catalog.has_admin_access'
+PERMISSION_HAS_CATALOG_PROVISIONING_ADMIN_ACCESS = 'provisioning.has_catalog_provisioning_admin_access'
```
Comment (sameenfatima78) [inline enterprise_catalog/apps/catalog/constants.py]: Can we rename this to "catalog.has_provisioning_admin_access"?

File: enterprise_catalog/apps/catalog/constants.py [hunk 3]
```diff
@@ -63,16 +63,19 @@
 
 ENTERPRISE_CATALOG_ADMIN_ROLE = 'enterprise_catalog_admin'
 ENTERPRISE_CATALOG_LEARNER_ROLE = 'enterprise_learner'
+ENTERPRISE_CATALOG_PROVISIONING_ADMIN_ROLE = 'enterprise_catalog_provisioning_admin_role'
```
Comment (sameenfatima78) [inline enterprise_catalog/apps/catalog/constants.py]: nit: Remove role from `enterprise_catalog_provisioning_admin_role`

File: enterprise_catalog/apps/catalog/constants.py (hunk 2 above)
Comment (sameenfatima78) [reply enterprise_catalog/apps/catalog/constants.py]: Also, rename `PERMISSION_HAS_CATALOG_PROVISIONING_ADMIN_ACCESS` to `PERMISSION_HAS_PROVISIONING_ADMIN_ACCESS`

**Review by sameenfatima78 — APPROVED** [review]

File: enterprise_catalog/apps/catalog/rules.py (hunk 1 above)
Comment (hamzawaleed01) [reply enterprise_catalog/apps/catalog/rules.py]: For PAs we're relying on role information in JWT that's why checking for implicit access only is sufficient.

File: enterprise_catalog/apps/api/v1/tests/test_views.py [hunk 4]
```diff
@@ -328,7 +326,7 @@ def test_patch_provisioning_admins(self):
         self.set_up_staff_user()
```
Comment (zamanafzal) [inline enterprise_catalog/apps/api/v1/tests/test_views.py]: Do we need to test it without staff user?

Comment (hamzawaleed01) [reply enterprise_catalog/apps/api/v1/tests/test_views.py]: Thanks for pointing it out! yeah it was already a non-staff user but was unnecessarily added. Line #264, #265 and removing its role assignments.

**Pull request merged**